cachi2/core/package_managers/cargo/main.py

```diff
@@ -34,7 +36,17 @@ def _resolve_cargo_package(package_dir: RootedPath, output_dir: RootedPath) -> l
     cmd = ["cargo", "vendor", "--locked", str(vendor_dir)]
     log.info("Fetching cargo dependencies at %s", package_dir)
     run_cmd(cmd=cmd, params={"cwd": package_dir})
-    return []
+
+    downloaded_deps = os.listdir(vendor_dir)
+    all_configs = [vendor_dir.path / x / "Cargo.toml" for x in downloaded_deps]
+    all_tomls = [TOMLFile(x).read() for x in all_configs]
+    # Ignoring mypy error since name and version will be present in all Cargo.tomls downloaded
+    # from a registry.
+    name_versions = [(x["package"]["name"], x["package"]["version"]) for x in all_tomls]  # type: ignore
+    mk_purl = lambda n, v: PackageURL(type="cargo", name=n, version=v).to_string()  # noqa: E731
+    mk_comp = lambda n, v: Component(name=n, version=v, purl=mk_purl(n, v))  # noqa: E731
```

Comment:
This doesn't seem to be aligned with how other backends produce the information. I know in case of cargo most of the job has been done by cargo itself via vendoring, but I think **consistency** is key regardless, i.e. following the precedent set by other modules (i.e. using `purl` and `to_component` instance methods) for the small tradeoff of defining tiny dataclasses and some negligible overhead I think.

Reply:
If we don't have it even for yarn which is the most recent example, then well, we should be more diligent about it :) .